src/App.tsx

```diff
@@ -1,26 +1,259 @@
+          setLoadingError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingTodos(prev => prev.filter(todoId => todoId !== id));
+      });
+  };
+
+  const updatedPost = (todo: Todo) => {
+    setLoadingTodos(prev => [...prev, todo.id]);
+
+    updatedTodo(todo)
+      .then(() => {
+        setTodos(prevTodos =>
+          prevTodos.map(t => (t.id === todo.id ? todo : t)),
+        );
+      })
+      .catch(() => {
+        setLoadingError('Unable to update a todo');
+        setTimeout(() => {
+          setLoadingError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingTodos(prev => prev.filter(todoId => todoId !== todo.id));
+      });
+  };
+
+  const clearErrors = () => {
+    setLoadingError('');
+    setValidationError('');
+  };
+
+  const clearCompleted = () => {
+    const completedTodoIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    completedTodoIds.forEach(id => {
+      deletePost(id);
+    });
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const leftItem = todos.filter(todo => !todo.completed).length;
+  const disabledButton = todos.every(todo => !todo.completed);
+
+  const filterValues = Object.values(Filter);
+
+  const allTodos = [...filterTodos, ...tempTodos];
+
+  const cnButton = todos.every(todo => todo.completed);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={cn('todoapp__toggle-all', { active: cnButton })}
+            data-cy="ToggleAllButton"
+            onClick={toggleTodoAll}
+          />
+          <form onSubmit={handleAddTodo}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              ref={inputRef}
+              value={inputText}
+              onChange={handleInputChange}
+              disabled={loadingTodos.length > 0}
+            />
+          </form>
+        </header>
+        <TodoList
+          todos={allTodos}
+          toggleTodo={toggleTodo}
+          deletePost={deletePost}
+          loadingTodos={loadingTodos}
+          updatedPost={updatedPost}
+        />
+        {todos.length > 0 && (
+          <Footer
+            leftItem={leftItem}
+            filterValues={filterValues}
```

Comment:
it's static value - no need to pass it as prop
just import Filter enum in the Footer componet